SmartSearch.LuceneNet/Internals/Converters/IndexableFieldConverter.Types.cs

```diff
@@ -14,47 +14,71 @@ namespace SmartSearch.LuceneNet.Internals.Converters
 {
     internal interface ITypedIndexableFieldConverter
     {
-        IEnumerable<IIndexableField> Convert(InternalSearchDomain domain, IField field, InternalDocument sourceDocument);
+        IEnumerable<IIndexableField> Convert(
+            InternalSearchDomain domain,
+            IField field,
+            InternalDocument sourceDocument
+        );
     }
 
     internal class BoolIndexableFieldConverter : TypedSimpleOrArrayIndexableFieldConverterBase
     {
-        public BoolIndexableFieldConverter() : base(FieldType.Bool, FieldType.BoolArray)
-        {
-        }
-
-        protected override IIndexableField GetIndexableField(InternalSearchDomain domain, IField field, object value) =>
-            new Int32Field(field.Name, BoolConverter.ConvertToInt(value), GetFieldStore(field)) { Boost = GetFieldBoost(field) };
+        public BoolIndexableFieldConverter() : base(FieldType.Bool, FieldType.BoolArray) { }
+
+        protected override IIndexableField GetIndexableField(
+            InternalSearchDomain domain,
+            IField field,
+            object value
+        ) =>
+            new Int32Field(field.Name, BoolConverter.ConvertToInt(value), GetFieldStore(field))
+            {
+                Boost = field.RelevanceBoost
```

Comment:
The Relevance Boost is no longer calculated on-demand here; it is now set to the field when it is created.